gusto/diagnostics.py

```diff
@@ -1581,3 +1582,34 @@ def setup(self, domain, state_fields):
             state_fields (:class:`StateFields`): the model's field container.
         """
         super().setup(domain, state_fields, vorticity_type="relative")
+
+
+class TracerDensity(DiagnosticField):
+    """Diagnostic for computing the density of a tracer. This is
+    computed as the product of a mixing ratio and dry density"""
+
+    name = "TracerDensity"
+
+    def __init__(self, m_X, rho_d):
+        """
+        Args:
+            m_X: the mixing ratio of the tracer
```

Comment:
We need to include the type of the argument in the docstrings, so this should be something like:
```
mixing_ratio_name (str), the name of the tracer mixing ratio variable.
```
and so on for the other variables